samza-api/src/main/java/org/apache/samza/context/JobContext.java

```diff
@@ -0,0 +1,15 @@
+package org.apache.samza.context;
+
+import org.apache.samza.config.Config;
+
+
+/**
+ * Contains information at job granularity, provided by the Samza framework, to be used to instantiate an application at
+ * runtime.
+ */
+public interface JobContext {
+  /**
+   * @return final configuration for this job
+   */
+  Config getConfig();
```

Comment:
Maybe expose JobName, JobId as top level params for consistency with other contexts.

Also, don't we want to expose JobModel here?

Reply:
Added getJobName and getJobId